### PR TITLE
plugins ar_spectral: fix spectrum being inverted on some systems

### PR DESCRIPTION
**Pull request by pieleric**
We use the raw CCD, although normally it's the spectrometer, which wraps
the CCD that has an axis invertion if the wavelengths are the opposite
direction. Due to difference between hardware and possible extra mirror,
there is no convention of wavelength on the CCD. In practice this means
that there is 50% chance the wavelengths are inverted.

=> Automatically look at the spectrometer, and invert the wavelengths if
it does that.

Also remove some incorrect metadata which caused the data to be
sometimes not properly be displayed on the spacial image.

### CODE REVIEW

File: plugins/ar_spectral.py [hunk 1]
```diff
@@ -803,15 +819,35 @@ def __init__(self, microscope, main_app):
...
+        # in order to make sure the wavelength is is the correct direction (ie,
+        # lowest pixel = lowest wavelength). So we need to do the same on the
+        # raw image. However, there is no "official" way to connect the
+        # spectrometer(s) to their raw CCD. So we rely on the fact that
+        # typically this is a wrapper, so we can check using the .dependencies.
+        wl_inverted = False
+        try:
+            spec = self._find_spectrometer(self.ccd)
+        except LookupError as ex:
+            logging.warning("%s, expect that the wavelengths are not inverted", ex)
```
Comment (VictoriaDel) [inline plugins/ar_spectral.py:834]: This LookupError should be "Failed to find spectrometer... "

File: plugins/ar_spectral.py [hunk 2]
```diff
@@ -932,6 +968,22 @@ def _get_sem_survey(self):
         logging.warning("No SEM survey stream found")
         return None
 
+    def _find_spectrometer(self, detector):
+        """
+        Find a spectrometer which wraps the given detector
+        return (Detector): the spectrometer
+        raise LookupError: if nothing found.
+        """
+        for spec in self.main_app.main_data.spectrometers:
```
Comment (VictoriaDel) [inline plugins/ar_spectral.py:977]: You could replace main_app.main_data with main_data, as you did earlier for getting self._acqui_tab.

File: plugins/ar_spectral.py [hunk 3]
```diff
@@ -932,6 +968,22 @@ def _get_sem_survey(self):
...
+        Find a spectrometer which wraps the given detector
+        return (Detector): the spectrometer
+        raise LookupError: if nothing found.
+        """
+        for spec in self.main_app.main_data.spectrometers:
+            if not model.hasVA(spec, "dependencies"):
+                continue
+            # Check by name as the components are actually Pyro proxies, which
+            # might not be equal even if they point to the same component.
+            if detector.name in (d.name for d in spec.dependencies.value):
```
Comment (VictoriaDel) [inline plugins/ar_spectral.py:982]: Only one if condition with "and" would be clearer to me :) But you know better.

File: plugins/ar_spectral.py [hunk 4]
```diff
@@ -803,15 +819,35 @@ def __init__(self, microscope, main_app):
...
+        try:
+            spec = self._find_spectrometer(self.ccd)
+        except LookupError as ex:
+            logging.warning("%s, expect that the wavelengths are not inverted", ex)
+        else:
+            # Found spec => check transpose in X (1 or -1), and invert if it's inverted (-1)
+            try:
+                wl_inverted = (spec.transpose[0] == -1)
+            except Exception as ex:
+                logging.warning("%s: expect that the wavelengths are not inverted", ex)
```
Comment (VictoriaDel) [inline plugins/ar_spectral.py:840]: We need to know if wl is inverted or not. Why would we get an exception? (just for warning the user?)

**Review by VictoriaDel — CHANGES_REQUESTED** [review]

File: plugins/ar_spectral.py (hunk 1 above)
Comment (pieleric) [reply plugins/ar_spectral.py:834]: Yes, indeed. So the warning will look like "Failed to find spectrometer... , expect that the wavelengths...". It's fine, right?

**Review by pieleric — COMMENTED** [review]
I've fixed some of the comments, and answered back for the ones I couldn't fix.

**Review by VictoriaDel — APPROVED** [review]
I think only the warning log of the LookupError remains to be updated, and it's fine :)

File: plugins/ar_spectral.py (hunk 1 above)
Comment (pieleric) [reply plugins/ar_spectral.py:834]: Yes, that's fine I think.

**Pull request merged**